src/lib/qnio/nio_server.c

```diff
@@ -412,6 +430,17 @@ qns_server_start(char *node, char *port)
                         nioDbg("Accepted connection on descriptor %d "
                                   "(host=%s, port=%s)\n", infd, hbuf, sbuf);
                     }
+                    if(cmn_ctx->ssl_ctx)
+                    {
+                        ssl = SSL_new(cmn_ctx->ssl_ctx);
+                        SSL_set_fd(ssl, infd);
+                        ret = SSL_accept(ssl);
```

Comment:
This is blocking.  A client can DoS the server by never completing the handshake.  No other clients can connect during this time.  This needs to be non-blocking.